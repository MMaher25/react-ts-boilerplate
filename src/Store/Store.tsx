import React, { createContext, useReducer } from 'react'
import { IState } from '../Interfaces/Interfaces'
import { reducer } from '../Reducers/Reducers'

export const initialState: IState = {
  pokemon: {
    abilities: [],
    base_experience: 0,
    forms: [],
    game_indices: [],
    height: 0,
    held_items: [],
    id: 0,
    is_default: false,
    location_area_encounters: '',
    moves: [],
    name: '',
    order: 0,
    species: [],
    sprites: [],
    stats: [],
    types: [],
    weight: 0,
  }
}

export const Store = createContext<IState | any>(initialState)

export const StoreProvider = ({ children }: JSX.ElementChildrenAttribute): JSX.Element => {
  const [state, dispatch] = useReducer(reducer, initialState)
  return <Store.Provider value={{ state, dispatch }}>{children}</Store.Provider>
}
