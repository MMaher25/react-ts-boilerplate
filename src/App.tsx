import React, { Fragment, useContext, useEffect } from 'react'
import { hot } from 'react-hot-loader/root';
import { Store } from './Store/Store'
import { fetchDataAction } from './Actions/Actions'
import Pokemon from './Components/Pokemon'


const App: React.FC = (): JSX.Element => {
  const { state, dispatch } = useContext(Store)

  useEffect(() => {
    state.pokemon.name === '' && fetchDataAction(dispatch)
  })

  return (
    <Fragment>
      <Pokemon />
    </Fragment>
  )
}

export default hot(App);
