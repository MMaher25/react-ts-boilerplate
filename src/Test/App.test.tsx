import React from 'react';
import App from '../App';
import { StoreProvider } from '../Store/Store';
import { create } from 'react-test-renderer';

describe("App component", () => {
  test("Matches the snapshot", () => {
    const app = create(<StoreProvider><App /></StoreProvider>);
    expect(app.toJSON()).toMatchSnapshot();
  });
});
