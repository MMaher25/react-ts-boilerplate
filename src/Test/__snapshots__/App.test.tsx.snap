// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Matches the snapshot 1`] = `
<div
  className="flex-container"
>
  <div
    className="flex-item"
  >
    <div>
      <img
        id="image"
      />
    </div>
    <div
      id="pokemon-name"
    >
      
       #
      0
    </div>
    <button
      onClick={[Function]}
    >
      Get Random Pokemon
    </button>
  </div>
</div>
`;
